src/installer.py

```diff
@@ -1,194 +1,210 @@
 import subprocess
 import logging
 import os
+import pathmod
+import tempfile
 # Written Modules
 from utils import cmd
 
-def install_repos(mname, cname, repos, repo_dest, pkg_man, proxy):
-    # check if there are repos passed for install
-    if len(repos) == 0:
-        logging.info("REPOS: no repos passed to install\n")
-        return
+class Installer:
+    def __init__(self, pkg_man, cname, mname):
+        self.pkg_man = pkg_man
+        self.cname = cname
+        self.mname = mname
+
+        # Create temporary directory for logs, cache, etc. for package manager
+        os.makedirs(os.path.join(mname, "tmp"))
+        self.tdir = tempfile.mkdtemp(dir=os.path.join(self.mname, "tmp"), prefix="image-build-")
```

Comment:
Right now the tempdir is being created within the container being created, which gets umounted/deleted if errors occur. It just occurred to me that it might be useful to examine package manager logs should something go wrong and so it might make more sense to create the tempdir outside of the container on the system (whether host system or container) where `image-build` is being run. That way, the logs could be examined even if errors occur and the container that `image-build` creates gets deleted.